spa/spa.go

```diff
@@ -0,0 +1,48 @@
+package spa
+
+import (
+	"embed"
+	"errors"
+	"fmt"
+	"io/fs"
+	"net/http"
+
+	"github.com/NYTimes/gziphandler"
+)
+
+// Init creates a single page application manager.
+func Init(build embed.FS, dir string, index string) (*SPA, error) {
+	fsys, err := fs.Sub(build, dir)
+	if err != nil {
+		panic(fmt.Errorf("couldn't create sub filesystem: %w", err))
+	}
+
+	_, err = fsys.Open("index.html")
+	if err != nil {
+		if errors.Is(err, fs.ErrNotExist) {
+			return nil, fmt.Errorf("ui is enabled but no index.html found: %w", err)
+		} else {
+			return nil, fmt.Errorf("ui assets error: %w", err)
+		}
+	}
+
+	return &SPA{&RouterFS{index: index, FS: http.FS(fsys)}}, nil
+}
+
+type SPA struct {
```

Comment:
why do we need this type?

we could just have a single function: `spa.Handler(build embed.FS, dir, index string, useGzip bool) (http.Handler, error)`  right? and hide (private) everything else.

Reply:
I kept this only with the thought that extension of the method will be easy if needed. similar to gzip handler etc.

Reply:
since the struct is not really sharing state that is used repeatedly, i think just having `Handler() (http.Handler, error)`, `GZipHandler() (http.Handler, error)` etc as separate functions or `Handler(..., useGZip bool) (http.Handler, error)` is better. (overall less public surface area for the package and less things to understand before being able to use it).

Reply:
Yeah, we can do that as well. Changing it.